packages/lib/src/vis/shared/XAxisZoomMesh.tsx

```diff
@@ -0,0 +1,39 @@
+import { useThree } from '@react-three/fiber';
+import { Vector3 } from 'three';
+
+import type { ModifierKey } from '../models';
+import { useZoomOnWheel } from './hooks';
+
+const ZOOM_FACTOR = 0.95;
+
+interface Props {
+  disabled?: boolean;
+  modifierKey?: ModifierKey;
+}
+
+function XAxisZoomMesh(props: Props) {
+  const { disabled, modifierKey = 'Alt' } = props;
+
+  const { width, height } = useThree((state) => state.size);
+
+  const zoomVector = (sourceEvent: WheelEvent) => {
+    const factor = sourceEvent.deltaY > 0 ? ZOOM_FACTOR : 1 / ZOOM_FACTOR;
+
+    return new Vector3(
+      sourceEvent.getModifierState(modifierKey) ? 1 / factor : 1,
+      1,
+      1
+    );
+  };
+
+  const onWheel = useZoomOnWheel(zoomVector, disabled);
+
+  return (
+    <mesh onWheel={onWheel}>
+      <meshBasicMaterial opacity={0} transparent />
+      <planeGeometry args={[width, height]} />
+    </mesh>
+  );
```

Comment:
Can be left to another PR, but I feel like this could be abstracted to a `CanvasMesh` of sorts.

Reply:
Abstracted in `InteractionMesh`